### PR TITLE
Extra functionality for support external members [19710]

### PR DESCRIPTION
**Pull request by richiware**
Depends on:
- eprosima/fast-cdr#170

### CODE REVIEW

File: src/main/java/com/eprosima/idl/parser/typecode/MemberedTypeCode.java [hunk 1]
```diff
@@ -91,8 +91,62 @@ public List<Member> getMembers()
...
+        if (member.isAnnotationKey() && Kind.KIND_STRUCT != getKind())
+        {
+            throw new ParseException(null, "Error in member " + member.getName() +
+                    ": @key annotations only supported for structure's members.");
+        }
+        if (null != member.getAnnotationBitBound() && (
+                    Kind.KIND_ENUM != getKind() && Kind.KIND_BITMASK != getKind()))
+        {
+            throw new ParseException(null, "Error in member " + member.getName() +
+                    ": @bit_bound annotations only supported for structure's members or union's members.");
```
Comment (JLBuenoLopez-eProsima) [inline src/main/java/com/eprosima/idl/parser/typecode/MemberedTypeCode.java:127]:
```suggestion
                    ": @bit_bound annotations only supported for enumeration's members or bitmask's members.");
```

File: src/main/java/com/eprosima/idl/parser/typecode/MemberedTypeCode.java [hunk 2]
```diff
@@ -91,8 +91,62 @@ public List<Member> getMembers()
         return new ArrayList<Member>(m_members.values());
     }
 
-    public boolean addMember(Member member)
+    public boolean addMember(Member member) throws ParseException
```
Comment (JLBuenoLopez-eProsima) [inline src/main/java/com/eprosima/idl/parser/typecode/MemberedTypeCode.java:94]: I am missing the following check: 

> XTypes v1.3 Clause 7.3.1.2.1.2 & 7.3.1.2.1.3 It is an error to declare the same member as both optional and as a key.

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: src/main/java/com/eprosima/idl/parser/typecode/MemberedTypeCode.java (hunk 2 above)
Comment (richiware) [reply src/main/java/com/eprosima/idl/parser/typecode/MemberedTypeCode.java:94]: Done in 94abe2c

File: src/main/java/com/eprosima/idl/parser/typecode/MemberedTypeCode.java (hunk 1 above)
Comment (richiware) [reply src/main/java/com/eprosima/idl/parser/typecode/MemberedTypeCode.java]: Done in 94abe2c

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM

**Pull request merged**